live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/ApplicationInfoProvider.java

```diff
@@ -59,7 +63,13 @@ public String getUserAgentData()
     private String calculateUserAgentData()
     {
         String applicationVersion = integrationProperties.application().version();
-        String repositoryVersion = discoveryApiClient.getRepositoryVersion();
+        String repositoryVersion = Optional.of(integrationProperties)
+                .map(IntegrationProperties::alfresco)
+                .map(IntegrationProperties.Alfresco::repository)
+                .map(Repository::versionOverride)
+                .filter(Predicate.not(String::isBlank))
+                .orElseGet(() -> discoveryApiClient.getRepositoryVersion()
+                        .orElseThrow(() -> new IllegalStateException("The repository version cannot be retrieved from either the Discovery API or the Live Ingester configuration.")));
```

Comment:
FWIW I think you can just keep it all inline:
```suggestion
                .orElseGet(() -> discoveryApiClient.getRepositoryVersion())
                .orElseThrow(() -> new IllegalStateException("The repository version cannot be retrieved from either the Discovery API or the Live Ingester configuration."));
```

Reply:
We cannot implement it in this way because `orElseGet` returns `String` not `Optional<String>` so the suggested change won't compile.